python/generate_dist.py

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import pathlib
+import shutil
+
+
+def main():
+    src_dir = pathlib.Path(os.environ["MESON_SOURCE_ROOT"]).parent.resolve()
+    dist_dir = pathlib.Path(os.environ["MESON_DIST_ROOT"]).resolve()
+    subproj_dir = dist_dir / "subprojects" / "arrow-nanoarrow"
+
+    if subproj_dir.is_symlink():
+        subproj_dir.unlink()
+
+    subproj_dir.mkdir()
+    shutil.copy(src_dir / "meson.build", subproj_dir / "meson.build")
+    shutil.copy(src_dir / "meson.options", subproj_dir / "meson.options")
+
+    subproj_subproj_dir = subproj_dir / "subprojects"
+    subproj_subproj_dir.mkdir()
+    for f in (src_dir / "subprojects").glob("*.wrap"):
+        shutil.copy(f, subproj_subproj_dir / f.name)
+
+    target_src_dir = subproj_dir / "src"
+    shutil.copytree(src_dir / "src", target_src_dir)
+
+    # this files are only needed by bootstrap.py
```

Comment:
The amount of scripting here is unfortunate...would still be nice to replace bootstrap.py, but not sure I want to tackle that in this PR

Reply:
What do you mean by "including a meson lib definition?"

Reply:
Ah OK - so you want to run bootstrap.py continuously, not just when the distribution is being created?

The current form uses the subproject symlink for live integration with the C project and only vendors when the distribution is created

Reply:
OK I understand your point of view looking at this some more. The bootstrapping mechanism is pretty tightly engrained as is, and would be an effort in and of itself to replace.

Still looking at the right balance here - might take me a bit